optimade/server/entry_collections/entry_collections.py

```diff
@@ -255,6 +255,8 @@ def all_fields(self) -> Set[str]:
             # All provider-specific fields
             self._all_fields |= {
                 f"_{self.provider_prefix}_{field_name}"
+                if not field_name.startswith("_")
```

Comment:
I was wondering could some databases not already have internal field names starting with an underscore, but not a valid prefix ?
In that case, it would be better to check if a whole prefix is present.

Reply:
They could, but unless they define the field in their config then it won't show up anyway. Either way, they should be able to access those fields (they cannot currently) and can configure a mapper to choose a better field name.

Reply:
Yes that is true, I think I used a field name with an underscore to transfer information that was relevant for processing an entry but not useful to share with the users.